tests/examples/awful/template.lua

```diff
@@ -154,6 +159,24 @@ local function client_widget(c, col, label)
 end
 
 -- Add all wiboxes
+
+-- Fix the wibox geometries that have a dependency on their content
+for _, d in ipairs(drawin.get()) do
+    local w = d.get_wibox and d:get_wibox() or nil
+    if w then
+        -- Force a full layout first as widgets with as the awful.popup have
+        -- interdependencies between the content and the container
+        if w._apply_size_now then
+            w:_apply_size_now()
+        end
+    end
+end
+
+-- Emulate the event loop for another 5 iterations
+for _ = 1, 5 do
```

Comment:
The "right" number of probably 3 or 4. It is hit by `wiboxtypes.lua` from the popup PR. In no particular order: One of them is the first delayed call for layouts. One is `arrange`. One is the popup moving because it got resized. Another is the tooltip/popup attached to the popup that just moved. Then there is the drawing of the popup that just moved.

So 10 is currently safe for the current and soon to exist tests. That semi random number can be increased if the need arise in the future.

Reply:
Why do layouts need their own delayed call? Isn't this why `arrange` already does one?

Why does the popup not simply move immediately when it is resized? Apparently it was able to figure out its position before, so it should still be able to do this. Also, this doesn't sound like a particular expensive operation.

For the tooltip, I am not quite sure what is going on, but it should also be possible to move immediately.

Oh and: Why does this even need a loop? `gears.timer` runs delayed calls that are added during a delayed call in the same "iteration of the `refresh` signal". Thus, all but the first delayed call should be no-ops.

Reply:
https://github.com/awesomeWM/awesome/blob/master/lib/wibox/drawable.lua#L396
https://github.com/awesomeWM/awesome/blob/master/lib/awful/layout/init.lua#L199

Those 2 cause their respective layouts `refresh` requirement.

Reply:
> gears.timer runs delayed calls that are added during a delayed call in the same "iteration of the refresh signal". Thus, all but the first delayed call should be no-ops.

Or more explicitly, the following will print `1`, `2`, `3` (and not `1`, `3`, `2`):
```lua
gears.timer.delayed_call(function()
  print("1")
  gears.timer.delayed_call(function()
    print("2")
  end)
end)
awesome.emit_signal("refresh")
print("3")
```